harness/go/harness/util.go

```diff
@@ -13,3 +17,31 @@ var RetryDisabled = &temporal.RetryPolicy{MaximumAttempts: 1}
 func AppErrorf(msg string, args ...interface{}) error {
 	return temporal.NewApplicationError(fmt.Sprintf(msg, args...), "SdkFeaturesError")
 }
+
+type JSONPayload struct {
+	Data     string            `json:"data"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+func UnmarshalPayload(data []byte) (*JSONPayload, error) {
```

Comment:
Mentioned above, but consider using proto JSON unmarshalling instead of this custom object